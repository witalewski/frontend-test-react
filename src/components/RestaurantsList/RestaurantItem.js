import React from 'react';
import { bool, number, shape, string } from 'prop-types';
import { Link } from 'react-router-dom';
import { PropTypes } from 'mobx-react';
import { Status } from '../Status';
import { StarRating } from '../StarRating';

const RestaurantItem = ({
  restaurant: { name, image_url, rating, categories, price, is_closed },
}) => {
  const category = categories.length && categories[0].title;

  return (
    <div className="restaurants-list-item">
      <img alt={name} src={image_url} />
      <h2>{name}</h2>
      <StarRating rating={rating} />
      <div className="restaurants-list-item__details">
        <div>
          {category} â€¢ {price}
        </div>
        <Status open={!is_closed} />
      </div>
      <div className="restaurants-list-item__learn-more-wrapper">
        <Link to={`/${name.replace(/\W/g,'-')}`}>Learn more</Link>
      </div>
    </div>
  );
};

RestaurantItem.propTypes = {
  restaurant: shape({
    name: string,
    image_url: string,
    rating: number,
    categories: PropTypes.arrayOrObservableArrayOf(
      shape({
        title: string,
      })
    ),
    price: string,
  }).isRequired,
  is_closed: bool,
};

export { RestaurantItem };
