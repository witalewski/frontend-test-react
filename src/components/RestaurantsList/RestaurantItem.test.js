import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MOCK_RESTAURANT from '../../fixtures/restaurant.json';
import { RestaurantItem } from './RestaurantItem';

describe('RestaurantItem', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<RestaurantItem restaurant={MOCK_RESTAURANT} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
