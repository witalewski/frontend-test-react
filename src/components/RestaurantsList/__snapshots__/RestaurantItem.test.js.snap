// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantItem matches snapshot 1`] = `
<div
  className="restaurants-list-item"
>
  <img
    alt="Bagels & Friends"
    src="https://s3-media2.fl.yelpcdn.com/bphoto/zmRMxckIj4rhutGrSRcvYA/o.jpg"
  />
  <h2>
    Bagels & Friends
  </h2>
  <StarRating
    rating={4.5}
  />
  <div
    className="restaurants-list-item__details"
  >
    <div>
      Breakfast & Brunch
       â€¢ 
      $$
    </div>
    <Status
      open={true}
    />
  </div>
  <div
    className="restaurants-list-item__learn-more-wrapper"
  >
    <Link
      replace={false}
      to="/restaurant/wPSiShW_YSwlyKAJJUZ4fQ/Bagels---Friends"
    >
      Learn more
    </Link>
  </div>
</div>
`;
